addons/mass_mailing/models/mail_mail.py

```diff
@@ -88,12 +88,14 @@ def send_get_mail_body(self, cr, uid, mail, partner=None, context=None):
 
     def send_get_email_dict(self, cr, uid, mail, partner=None, context=None):
         res = super(MailMail, self).send_get_email_dict(cr, uid, mail, partner, context=context)
+        base_url = self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url')
         if mail.mailing_id and res.get('body') and res.get('email_to'):
             emails = tools.email_split(res.get('email_to')[0])
             email_to = emails and emails[0] or False
-            unsubscribe_url = self._get_unsubscribe_url(cr, uid, mail, email_to, context=context)
-            if unsubscribe_url:
-                res['body'] = tools.append_content_to_html(res['body'], unsubscribe_url, plaintext=False, container_tag='p')
+            unsubscribe_url, msg = self._get_unsubscribe_url(cr, uid, mail, email_to, context=context)
+            link_to_replace =  base_url+'/unsubscribe_from_list'
```

Comment:
Because we are giving Link to unsubscribe In each footer snippet of all the themes so we have to set the proper href there. And we can't have unsubscribe link at the time of designing. It will be generated while sending the mails (link needs recordset  of created mail that we won't be having while the time of designing).